templates/pathRouting.tmpl

```diff
@@ -0,0 +1,40 @@
+//this file is auto generated
+
+package generated
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+{{range $}}
+    {{range .Handlers}}
+        var I{{.Name}} {{.Name}}
+    {{end}}
+{{end}}
+
+func CreateAPIRouter() *mux.Router {
+	router := mux.NewRouter()
+	router.KeepContext = true
+
+{{range $}}
+	router.HandleFunc("{{.Path}}", func(res http.ResponseWriter, req *http.Request) {
+        // need to decide which handler to call here
```

Comment:
This is what I'm fighting with now - need to examine req here and decide which handler to call